onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -235,6 +236,7 @@ InlinedVector<std::unique_ptr<GraphTransformer>> GenerateTransformers(
       transformers.emplace_back(std::make_unique<EmbedLayerNormFusion>(cpu_cuda_rocm_eps));
 
       transformers.emplace_back(std::make_unique<BiasDropoutFusion>(cuda_rocm_eps));
+      transformers.emplace_back(std::make_unique<BitmaskDropoutReplacement>(cuda_rocm_eps));
```

Comment:
I still have to add it here because this is the only place that has EP info set (after InferenceSession init). The place for training is actually before InferenceSession init, which doesn't have the EP info, so the optimizers won't take effect there.

Reply:
I am almost wondering if we should change some framework related code so that we run the training graph builder after the node placement is done. Maybe it will make things easier for us. But that is a discussion for another time. Thanks for the explanation.